users/vieting/jobs/fairseq_labels.py

```diff
@@ -0,0 +1,252 @@
+            print(f"Processing {corpus_path.get()}")
+            c = corpus.Corpus()
+            c.load(corpus_path.get())
+            out_trans_path = os.path.join(dest_path.get(), "transcriptions.txt")
+            for segment in c.segments():
+                audio_path = segment.recording.audio
+                audio_trans = segment.orth
+                assert os.path.exists(audio_path), f"Path {audio_path} does not exist."
+                # copy audio file
+                shutil.copy(audio_path, dest_path)
+
+                # write to transcription file
+                with open(out_trans_path, "a") as f:
+                    # TODO maybe can write whole path?
+                    audio_basename = os.path.basename(audio_path)
+                    f.write(f"{audio_basename} {audio_trans}\n")
+            print(f"Finished processing {corpus_path.get()}\n")
+        print("Finished processing all corpus files.")
+
+
+class FairseqFinetuneTaskCreationJob(Job):
+    """
+    Creates required task files for wav2vec finetuning with fairseq. This includes the following files:
+    - train.tsv
+    - train.ltr
+    - train.wrd
+    - valid.tsv
+    - valid.ltr
+    - valid.wrd
+    - dict.ltr.txt
+
+    For the script see https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/wav2vec_manifest.py for .tsv creation,
+    https://github.com/facebookresearch/fairseq/blob/91c364b7ceef8032099363cb10ba19a85b050c1c/examples/wav2vec/libri_labels.py aswell as
+    the issue https://github.com/facebookresearch/fairseq/issues/2493 for .wrd and .ltr creation.
+    """
+
+    def __init__(
+        self,
+        audio_dir_path,
+        file_extension="wav",
+        valid_percent=0.01,
+        seed=42,
+        path_must_contain=None
+    ):
+        """
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
+        :param str file_extension: file extension to look for in audio_dir_path
+        :param float valid_percent: percentage of files to be in validation set
+        :param int seed: random seed for splitting into train and valid set
+        :param str|None path_must_contain: if set, path must contain this substring
+            for a file to be included in the manifest
+        """
+        if isinstance(audio_dir_path, tk.Path):
+            self.audio_dir_paths = [audio_dir_path]
+        else:
+            assert isinstance(audio_dir_path, list)
+            self.audio_dir_paths = audio_dir_path
+        assert min([isinstance(path, tk.Path) for path in self.audio_dir_paths])
+
+        
+        self.manifest_audio_paths = None
+
+        self.file_extension = file_extension
+        self.valid_percent = valid_percent
+        assert 0.0 <= self.valid_percent <= 1.0
+        self.seed = seed
+        self.path_must_contain = path_must_contain
+
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
+
+        self.out_task_path = self.output_path("task", directory=True)
```

Comment:
Do you need that folder? Couldn't the files below just all be in the normal `output/` folder?

Reply:
How could I then "`tk.register()`" the whole task folder at once? My idea behind this `task/ `folder was to combine all files into one folder so that one can just do `tk.register()` this `task/` folder instead of all files separately. Would this be also possible when removing the `task/` folder and saving all the created files in the `output/` folder directly

Reply:
I see. I don't think you'd need to register it, but off the top of my head I don't know how you would refer to the folder in a clean way (like here `self.out_task_path`). So let's just keep it like that.